File: textMedley/compilers/javascript/std.js
function randomNum(max) {
  return Math.floor(Math.random() * max);
}

function __if__(cond, ifTrue, iFalse) {
  return parseFloat(cond) !== 0 && cond !== "" ? ifTrue : iFalse;
}

function __concat__(...params) {
  return params.join("");
}

function __plus__(num, numb) {
  return (parseFloat(num) + parseFloat(numb)).toString();
}

function __minus__(num, numb) {
  return (parseFloat(num) - parseFloat(numb)).toString();
}

function __times__(num, numb) {
  return (parseFloat(num) * parseFloat(numb)).toString();
}

function __divide__(num, numb) {
  return (parseFloat(num) / parseFloat(numb)).toString();
}

function __power__(num, numb) {
  return (parseFloat(num) ** parseFloat(numb)).toString();
}

function __modulo__(num, numb) {
  return (parseFloat(num) % parseFloat(numb)).toString();
}

function __floor__(num) {
  return Math.floor(parseFloat(num)).toString();
}

function __ceil__(num) {
  return Math.ceil(parseFloat(num)).toString();
}

function __clamp__(num, min, max) {
  return Math.min(
    parseFloat(max),
    Math.max(parseFloat(min), parseFloat(num))
  ).toString();
}

function __min__(num, numb) {
  return Math.min(parseFloat(num), parseFloat(numb)).toString();
}

function __max__(num, numb) {
  return Math.max(parseFloat(num), parseFloat(numb)).toString();
}

File: textMedley/compilers/ruby/std.rb
#TODO think why numbers are converted into floats instead of ints

def randomNum(max)
  rand(0...max)
end

def __if__(cond, ifTrue, ifFalse)
  (cond.to_f != 0 && cond != "") ? ifTrue : ifFalse
end

def __concat__(val, valb)
  val + valb
end

def __plus__(num, numb)
  (num.to_f + numb.to_f).to_s
end

def __minus__(num, numb)
  (num.to_f - numb.to_f).to_s
end

def __times__(num, numb)
  (num.to_f * numb.to_f).to_s
end

def __divide__(num, numb)
  (num.to_f / numb.to_f).to_s
end

def __power__(num, numb)
  (num.to_f ** numb.to_f).to_s
end

def __floor__(num)
  num.to_f.floor.to_s
end

def __ceil__(num)
  num.to_f.ceil.to_s
end

def __clamp__(num, min, max)
  [[num, min].max, max].min
end

def __min__(num, numb)
  [num, numb].min
end

def __max__(num, numb)
  [num, numb].max
end
